tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Test to restart ceph target service along with pwl cache test")
+
+    def restart_ceph_target():
+        cmd = "systemctl restart ceph.target"
+        admin_node.exec_command(cmd=cmd, sudo=True)
+
+    with parallel() as p:
+        p.spawn(restart_ceph_target)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def remove_ceph_service(service_obj, host, cache_client, pwl, config, image_spec):
+    """
+    Remove a Ceph service (e.g., Mon) from a host and verify the
+    persistent write cache in parallel.
+
+    Args:
+        service_obj: The Ceph service object (e.g., Mon, OSD).
+        host (str): The name of the host where the service will be removed.
+        cache_client (Node): The client node for cache testing.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict, optional): The test configuration. Default is None.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(
+        f"Test to remove ceph {service_obj.daemon_type} service along with pwl cache test"
+    )
+
+    def remove_service():
+        cmd = f"ceph orch host label rm {host} {service_obj.daemon_type}"
+        cache_client.exec_command(cmd=cmd, output=True)
+
+        if not service_obj.remove_mon_service(host=host):
+            log.error(f"Could not remove the {service_obj.daemon_type} host")
+            return 1
+        log.info(f"{service_obj.daemon_type} removed successfully")
+
+    with parallel() as p:
+        p.spawn(remove_ceph_service)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def add_ceph_service(service_obj, host, cache_client, pwl, config, image_spec):
+    """
+    Add a Ceph service (e.g., Mon) from a host and verify the
+    persistent write cache in parallel.
+
+    Args:
+        service_obj: The Ceph service object (e.g., Mon, OSD).
+        host (str): The name of the host where the service will be removed.
+        cache_client (Node): The client node for cache testing.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration. Default is None.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(
+        f"Test to add ceph {service_obj.daemon_type} service along with pwl cache test"
+    )
+
+    def add_service():
+        cmd = f"ceph orch host label add {host} {service_obj.daemon_type}"
+        cache_client.exec_command(cmd=cmd, sudo=True)
+        time.sleep(10)
+
+        if not service_obj.check_mon_exists_on_host(host=host):
+            log.error(f"Could not add {service_obj.daemon_type} service")
+            return 1
+        log.info(f"Added {service_obj.daemon_type} service successfully")
+
+    with parallel() as p:
+        p.spawn(add_service)
```

Comment:
This implementation still doesn't catch any return value and throw exception right? As we discussed.